### ISSUE TITLE
LT02 requires `FROM VALUES` to be on separate lines when the values are on separate lines

### ISSUE THREAD

**Issue by mroy-seedbox** [sqlfluff/sqlfluff#6675]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This should work:
```sql
SELECT *
FROM VALUES -- Expected line break and indent of 2 spaces before 'VALUES'. sqlfluff(LT02: layout.indent)
  (1, 'something'),
  (2, 'something');
```

But instead we must currently do this, which provides no benefit and looks strange:
```sql
SELECT *
FROM
  VALUES
  (1, 'something'),
  (2, 'something');
```

### Expected Behaviour

`FROM VALUES` on one line should be acceptable.

Especially with `allow_implicit_indents = True`.

### Observed Behaviour

`LT02` is triggered.

### How to reproduce

See above.

### Dialect

Snowflake

### Version

`3.3.1`

### Configuration

```
[sqlfluff]
templater = jinja
dialect = snowflake

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = False
indented_using_on = True
indented_ctes = False
indented_on_contents = False
allow_implicit_indents = True

[sqlfluff:rules:layout.spacing]
extra_whitespace = false

[sqlfluff:rules:structure.subquery]
forbid_subquery_in = both
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mroy-seedbox** [comment]
Related issues:
- #5855 (indentation problem for `INSERT INTO ... VALUES ...`)
- #5600 (support alias on `VALUES` clause)
- #3301 (support alias on `VALUES` clause)
- #3103 (support `VALUES` without surrounding parentheses)
